### PR TITLE
Advanced Fuel Calculations

### PR DESCRIPTION
**Pull request by FatihKoz**
PR aims to have realistic fuel debit calculations as in the real ops. When enabled, remaining fuel amounts from previous flight will be reduced from block fuel and airline will only pay for uplifted fuel amount.

If onboard fuel is enough for the flight or exceeds the required amount no fuel debit will be issued.

If this is the first flight of the aircraft or the tanks are dry, full block fuel will be treated as uplifted.

Disabled / uses current default (fuel_used will be debited)

Aircraft table also will hold the fuel_onboard value (not needed for calculations, just saving for displaying purposes)

### CODE REVIEW

File: app/Database/seeds/settings.yml [hunk 1]
```diff
@@ -305,3 +305,11 @@
...
   type: 'text'
   description: 'Discord public channel ID for broadcasat notifications'
+- key: pireps.advanced_fuel
+  name: 'Advanced Fuel Calculations'
+  group: pireps
+  value: false
+  options: ''
+  type: boolean
+  description: 'Enables remaining fuel amounts to be considered for fuel expenses'
+
```
Comment (nabeelio) [inline app/Database/seeds/settings.yml:315]: Move this so it's below where the other pirep settings are. And make sure there is a newline at the end of the file

Comment (FatihKoz) [reply app/Database/seeds/settings.yml:316]: Done

**Comment by nabeelio** [comment]
Great, looks like the tests passed now with the `settings()` fix - merged. Thanks!

**Comment by FatihKoz** [comment]
Yes, i was focused on EOF but it had a problem above too. Glad I saw it while checking the build errors. My pleasure and thanks for the quick merge.

**Pull request merged**